pkg/cmd/cli/backuplocation/delete.go

```diff
@@ -120,7 +121,65 @@ func Run(f client.Factory, o *cli.DeleteOptions) error {
 			continue
 		}
 		fmt.Printf("Backup storage location %q deleted successfully.\n", location.Name)
+
+		// Delete backups associated with the deleted BSL.
+		backupList, err := findAssociatedBackups(kbClient, location.Name, f.Namespace())
+		if err != nil {
+			errs = append(errs, fmt.Errorf("find backups associated with BSL %q: %w", location.Name, err))
+		}
+		if deleteErrs := deleteBackups(kbClient, backupList); deleteErrs != nil {
+			errs = append(errs, deleteErrs...)
+		}
+
+		// Delete Restic repositories associated with the deleted BSL.
+		resticRepoList, err := findAssociatedResticRepos(kbClient, location.Name, f.Namespace())
+		if err != nil {
+			errs = append(errs, fmt.Errorf("find Restic repositories associated with BSL %q: %w", location.Name, err))
+		}
+		if deleteErrs := deleteResticRepos(kbClient, resticRepoList); deleteErrs != nil {
```

Comment:
Should not call deleteResticRepos is we get an error from findAssociatedResticRepos because resticRepoList is probably bad.